Mathlib/Control/Random.lean

```diff
@@ -120,19 +130,26 @@ instance {n : Nat} : BoundedRandom (Fin n) where
 
 end Random
 
+open IO
+
+variable {m : Type _ → Type _} [Monad m] [MonadLiftT (ST RealWorld) m]
+
 /-- Computes a `Rand α` using the global `stdGenRef` as RNG.
     Note that:
     - `stdGenRef` is not necessarily properly seeded on program startup
       as of now and will therefore be deterministic.
     - `stdGenRef` is not thread local, hence two threads accessing it
       at the same time will get the exact same generator.
 -/
-def IO.runRand (cmd : Rand α) : BaseIO α := do
+def runRand (cmd : Rand m α) : m α := do
   let stdGen ← stdGenRef.get
   let rng := ULift.up stdGen
-  let (res, new) := Id.run <| StateT.run cmd rng
+  let (res, new) ← StateT.run cmd rng
   stdGenRef.set new.down
   pure res
 
-def IO.runRandWith (seed : Nat) (cmd : Rand α) : BaseIO α := do
-  pure $ (cmd.run (ULift.up $ mkStdGen seed)).1
+instance : MonadLift (Rand m) m where
+  monadLift := runRand
```

Comment:
This strikes me as a bit risky; I'd rather not accidentally run a random monad with the default generator and get something less deterministic than expecfted.